File: Lv1/하샤드수.java
package Lv1;

public class 하샤드수 {

    public boolean solution(int x) {

        String strX = String.valueOf(x);
        int sum = 0;
        for(int i = 0; i < strX.length(); i++) {
            sum += strX.charAt(i) - '0';
        }

        return x % sum == 0 ? true : false;
    }

}

File: kotlin/Lv1/하샤드수.kt
package kotlin.Lv1

class 하샤드수 {
    fun solution(x: Int): Boolean {
        var sum: Int = 0;
        val xStr = x.toString()

        for(i:Char in xStr.toCharArray()) {
            sum += (i - '0')
        }

        var answer = (x % sum == 0)
        return answer
    }
}